pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -684,8 +678,26 @@ func allocatableResources(memory, cpu string) v1.ResourceList {
 	}
 }
 
-// DaemonSets should not place onto nodes with insufficient free resource
+// When ScheduleDaemonSetPods is disabled, DaemonSets should not be placed onto nodes with insufficient free resource
+// We don't need this test as default scheduler is responsible for doing resource checks when ScheduleDaemonSetPods
+// is enabled. We are unsetting ScheduleDaemonSetPods to test codepath without ScheduleDaemonSetPods
+// TODO: Remove this test when ScheduleDaemonSetPods by default scheduler graduates to GA.
```

Comment:
We can only remove this test when `ScheduleDaemonSetPods` feature gate is removed. As long as cluster admins can disable this feature, we should continue support this path.